### PR TITLE
2FA setup wizard: Show private key

### PR DESCRIPTION
**Pull request by aph3rson**
Per #443, show the OTP secret/private key in the 2FA setup wizard.

A new variable is sent to the template, `secretGrouped`, a user-friendly format of that secret. This is then displayed in the wizard inside of a short sub-tutorial, which uses Bootstrap's collapsible elements to show only when a user clicks on the "Can't scan the barcode?" link.

When merged, closes pretix/pretix#443.

### CODE REVIEW

**Comment by aph3rson** [comment]
Also, apologies if this doesn't merge correctly - I used the wrong reverse PR to resync my repo to upstream.

File: src/pretix/control/templates/pretixcontrol/user/2fa_confirm_totp.html [hunk 1]
```diff
@@ -37,7 +37,38 @@
         </li>
         <li>
             {% trans "Add a new account to the app by scanning the following barcode:" %}
-            <div class="qrcode-canvas" data-qrdata="#qrdata"></div>
+          <div class="qrcode-canvas" data-qrdata="#qrdata"></div>
```
Comment (raphaelm) [inline src/pretix/control/templates/pretixcontrol/user/2fa_confirm_totp.html]: I'm not quite sure how it happened, but this indentation doesn't seem to be a multiple of four :wink:

File: src/pretix/control/templates/pretixcontrol/user/2fa_confirm_totp.html [hunk 2]
```diff
@@ -37,7 +37,38 @@
...
+            <p>
+              <a data-toggle="collapse" href="#no_scan" aria-expanded="false" aria-controls="collapseExample">
```
Comment (raphaelm) [inline src/pretix/control/templates/pretixcontrol/user/2fa_confirm_totp.html]: "collapseExample"? 😉 I think you can chop of the ``aria-`` attributes, we have them nowhere in pretix currently.

File: src/pretix/control/templates/pretixcontrol/user/2fa_confirm_totp.html [hunk 3]
```diff
@@ -37,7 +37,38 @@
...
+                {% trans "Can't scan the barcode?" %}
+              </a>
+            </p>
+          <div class="collapse" id="no_scan">
+            <ol class="multi-step-tutorial">
+              <li>
+                {% trans "Use the \"provide a key\" option of your authenticator app." %}
+              </li>
+              <li>
+                {% trans "In \"Account name\", type your login name for Pretix." %}
```
Comment (raphaelm) [inline src/pretix/control/templates/pretixcontrol/user/2fa_confirm_totp.html]: Lowercase "pretix"

File: src/pretix/control/templates/pretixcontrol/user/2fa_confirm_totp.html (hunk 1 above)
Comment (aph3rson) [inline src/pretix/control/templates/pretixcontrol/user/2fa_confirm_totp.html]: Fixed in 073c0d8 - not sure what Atom's issue with indentation was...

File: src/pretix/control/templates/pretixcontrol/user/2fa_confirm_totp.html (hunk 2 above)
Comment (aph3rson) [inline src/pretix/control/templates/pretixcontrol/user/2fa_confirm_totp.html]: Fixed in ba4b0c4 - wasn't sure how to use `aria-*` attributes, so I left them as is from the template... my bad!

File: src/pretix/control/templates/pretixcontrol/user/2fa_confirm_totp.html (hunk 3 above)
Comment (aph3rson) [inline src/pretix/control/templates/pretixcontrol/user/2fa_confirm_totp.html]: Fixed in 83b1956.

**Comment by aph3rson** [comment]
@raphaelm Fixed the things you mentioned - sorry for the tardy response!

**Comment by raphaelm** [comment]
Thank you very much! :tada:

If you want to continue contributing, you might want to have a look at the last open part of https://github.com/pretix/pretix/issues/102#issuecomment-300102157

**Pull request merged**